### DISCUSSION TITLE
Having node run conditionally?

### DISCUSSION THREAD

**Discussion by meaningfromdata** [kedro-org/kedro#1181] [Idea]
Sorry if I missed it in the docs but is there a way to conditionally run a node in a pipeline?  My use case is basically trying to do data version control where the first node of the pipeline reads in data from several tables in a database and hashes each of them (along the lines of DVC).  Subsequent nodes would only run if that data had changed from what had been seen previously (i.e. if any of the tables had a hash that hadn't been previously seen).

**Comment by datajoely** [comment]
Hi @meaningfromdata 

So this isn't something we have out of the box - the idea to include some sort of conditional node is one I've been thinking about for a while. We've also been exploring more comprehensive DVC integration, but it's a way off.

One way you could do it is to do an `before_node_run` [hook](https://kedro.readthedocs.io/en/stable/kedro.framework.hooks.specs.NodeSpecs.html#kedro.framework.hooks.specs.NodeSpecs.before_node_run).

- You could inspect the `inputs` dictionary, to verify if things have changed 
- Depending on the result of that check you could choose to continue or exit the programme. 

It's not the prettiest and for now you would have to come up with the logic for defining if the data is different.

**Reply by datajoely** [reply]
Read more about hooks here! https://kedro.readthedocs.io/en/latest/07_extend_kedro/02_hooks.html

**Reply by wmoreiraa** [reply]
Just a quick note, I've followed this suggestion in a similar workflow using DVC. Hook after_node_run worked fine.

**Reply by datajoely** [reply]
@wmoreiraa nice! Would you be interested in sharing a sample hook or project with the community?

**Reply by datajoely** [reply]
Maybe a `Show and Tell` discussion?
![image](https://user-images.githubusercontent.com/35801847/151328585-f4b513df-72a4-48e0-b270-f770d8902ff4.png)

**Reply by meaningfromdata** [reply]
@wmoreiraa Thanks for your reply.  Agree with @datajoely that would be interesting to see your DVC hook.

**Comment by FlyingDragon7777** [comment]


**Reply by datajoely** [reply]
Hi @FlyingDragon7777 I'm nots sure this is relevant to the discussion - was this a question about Kedro?

**Reply by FlyingDragon7777** [reply]
Not really, it was a general question

**Reply by FlyingDragon7777** [reply]
Thank you for replying so quickly

**Comment by dhavala** [comment]
That would be interesting. We were trying to integrate DVC with kedro.

**Comment by wmoreiraa** [comment]
Sure, guys!

@datajoely  Could we schedule a call early next week? Then we could discuss what we’re
aiming for and what we already have, then we could share with the community right after.

I will send you a DM with my e-mail

**Comment by datajoely** [comment]
There is a parallel conversation happening here too: https://github.com/kedro-org/kedro/discussions/837

**Comment by astrojuanlu** [comment]
Another ancient issue connected to this topic https://github.com/kedro-org/kedro/issues/104
